certbot/plugins/standalone.py

```diff
@@ -114,39 +112,53 @@ def running(self):
 SUPPORTED_CHALLENGES = [challenges.TLSSNI01, challenges.HTTP01]
 
 
-def supported_challenges_validator(data):
-    """Supported challenges validator for the `argparse`.
+class SupportedChallengesAction(argparse.Action):
+    """Action class for parsing standalone_supported_challenges."""
 
-    It should be passed as `type` argument to `add_argument`.
+    def __call__(self, parser, namespace, values, option_string=None):
```

Comment:
More `argparse` fun! As documented [here](https://docs.python.org/2.7/library/argparse.html#default), if the default is a `str` and you specify a `type` argument when adding the CLI flag to the parser, the `type` parameter gets called with the default before adding it to the namespace. As a result, sometimes when `type` was called we were just parsing the default rather than a user provided value so we had to use our `set_by_cli()` function to differentiate between these cases.

This avoids all that because the `action` argument is only called if the user provides a value.